src/services/dashboards/dashboard-customize/modules/DashboardAddDefaultWidgetRightArea.vue

```diff
@@ -108,20 +198,118 @@ export default defineComponent<Props>({
             name(value: string) { return value.trim().length ? '' : i18n.t('Please enter name.'); },
         });
 
+        const {
+            targetRef, targetElement, contextMenuStyle,
+        } = useContextMenuFixedStyle({
+            useFixedMenuStyle: true,
+            visibleMenu: toRef(state, 'addOptionsMenuVisible'),
+        });
+        const contextMenuFixedStyleState = reactive({
+            targetRef, targetElement, contextMenuStyle,
+        });
+
+        /* Util */
+        const isSelected = (selectedItem) => {
+            if (Array.isArray(selectedItem)) return !!selectedItem.length;
+            return selectedItem && !isEmpty(selectedItem);
+        };
+        const getRefinedJsonSchema = (originalJsonSchema: JsonSchema, selectedOptions, inheritItemMap) => {
+            const _jsonSchema = cloneDeep(originalJsonSchema);
+            const refinedJsonSchema: Partial<JsonSchema> = {
+                type: 'object',
+                properties: {},
+                required: state.requiredProperties,
+            };
+            Object.entries(_jsonSchema.properties).forEach(([propertyName, propertySchema]) => {
+                if (!state.requiredProperties.includes(propertyName) && !selectedOptions.find((d) => d.name === propertyName)) {
+                    return;
+                }
+                if (inheritItemMap[propertyName]) {
+                    refinedJsonSchema.properties[propertyName] = {
+                        title: (propertySchema as JsonSchema).title,
```

Comment:
question: why type assertion is needed?

Reply:
Without that, this error is occurred🥲 maybe there's no type of `properties`.

![스크린샷 2022-12-19 오후 1 43 21](https://user-images.githubusercontent.com/18563857/208349645-5aa8f917-13b0-4fa5-803e-413855f21250.png)

```typescript
export type JsonSchema<Properties = object> = JSONSchemaType<Properties> & {
    title?: string;
    order?: string[];
    disabled?: boolean;
    json?: boolean;
};
```